### ISSUE TITLE
Release 4.5.0 - Docker Listener Module testing

### ISSUE THREAD

**Issue by fdalmaup** [wazuh/wazuh#14416]
# Docker Listener module testing

## Modules tests information
|                                  |                                             |
|----------------------------------|---------------------------------------------|
| **Main module tests issue**      | #14376  |
| **Version**                      | 4.5.0                                       |
| **Release candidate #**          | -                                         |
| **Tag**                          | -|
| **Previous modules tests issue** | - |

## Test report procedure

All test results must have one the following statuses: 
|                                  |                                                        |
|-----------------|-------------------------------------------------------------------------|
| :green_circle:  | The test passed successfully. 										    |
| :yellow_circle: | The test passed but some enhancements need to be applied to the module. |
| :red_circle:    | The test failed.                                                        |
| :white_circle:  | The test does not apply to the given bucket type or service.            |

Any failing test must be properly addressed with a new issue, detailing the error and the possible cause. It must be included in the Release 4.5.0 project for triage.

The data available in our Wazuh-dev environment for each of these supported services will be used. Additional logs may be created if they are necessary or there is insufficient data for testing.

### Conclusions

| Test | Status Python 3.9 | Issue ref. |
|---|---|---|
| Testing docker-listener                             |   :green_circle:    |  -  |

## Auditors validation
The definition of done for this one is the validation of the conclusions and the test results from all auditors.

All checks from below must be accepted in order to close this issue.

- [ ] @davidjiglesias 
- [x] @CarlosRS9 
- [x] @fdalmaup

**Comment by fdalmaup** [comment]
## Tests

Manual testing was performed to ensure everything works after the change. The tests were carried out using both a Wazuh manager and a Wazuh agent both using the following configuration:
```
<wodle name="docker-listener">
    <interval>30s</interval>
    <attempts>5</attempts>
    <run_on_start>yes</run_on_start>
    <disabled>no</disabled>
</wodle>
```

The [logall_json option](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#logall-json) must be enabled in the `ossec.conf` file before starting the tests.

### Test cases Python 3.9

#### Restart Wazuh and check ossec.log to verify the module is running


<details><summary>ossec.log output</summary>

    2022/07/25 12:45:41 wazuh-modulesd[3056] main.c:95 at main(): DEBUG: Created new thread for the 'docker-listener' module.
    2022/07/25 12:45:41 wazuh-modulesd:docker-listener[3056] wm_docker.c:45 at wm_docker_main(): INFO: Module docker-listener started.
    2022/07/25 12:45:41 wazuh-modulesd:docker-listener[3056] wm_docker.c:58 at wm_docker_main(): INFO: Starting to listening Docker events.
    2022/07/25 12:45:41 wazuh-modulesd:docker-listener[3056] wm_docker.c:62 at wm_docker_main(): DEBUG: Launching command 'wodles/docker/DockerListener'
    
</details>



#### Test docker commands

**Run a container with no cache available**
```
docker run ubuntu
```

<details><summary>archives.json output</summary>

    {"timestamp":"2022-07-25T15:47:56.212+0000","rule":{"level":3,"description":"Docker: Image or repository ubuntu pulled","id":"87932","firedtimes":1,"mail":false,"groups":["docker"],"pci_dss":["10.2.7"],"hipaa":["164.312.b"],"nist_800_53":["AU.14"],"tsc":["CC6.8","CC7.2","CC7.3"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764076.1244457","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"pull\", \"id\": \"ubuntu:latest\", \"Type\": \"image\", \"Action\": \"pull\", \"Actor\": {\"ID\": \"ubuntu:latest\", \"Attributes\": {\"name\": \"ubuntu\"}}, \"scope\": \"local\", \"time\": 1658764076, \"timeNano\": 1658764076210417937}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"pull","id":"ubuntu:latest","Type":"image","Action":"pull","Actor":{"ID":"ubuntu:latest","Attributes":{"name":"ubuntu"}},"scope":"local","time":"1658764076","timeNano":"1658764076210417920.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T15:47:56.359+0000","rule":{"level":3,"description":"Docker: Container charming_euler created","id":"87901","firedtimes":1,"mail":false,"groups":["docker"],"pci_dss":["10.2.7"],"hipaa":["164.312.b"],"nist_800_53":["AU.14"],"tsc":["CC6.8","CC7.2","CC7.3"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764076.1245212","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"create\", \"id\": \"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"create\", \"Actor\": {\"ID\": \"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"charming_euler\"}}, \"scope\": \"local\", \"time\": 1658764076, \"timeNano\": 1658764076355395254}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"create","id":"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071","from":"ubuntu","Type":"container","Action":"create","Actor":{"ID":"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071","Attributes":{"image":"ubuntu","name":"charming_euler"}},"scope":"local","time":"1658764076","timeNano":"1658764076355395328.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T15:47:56.360+0000","rule":{"level":3,"description":"Docker: Attached local standard input, output, and error streams to container charming_euler","id":"87922","firedtimes":1,"mail":false,"groups":["docker"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764076.1246297","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"attach\", \"id\": \"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"attach\", \"Actor\": {\"ID\": \"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"charming_euler\"}}, \"scope\": \"local\", \"time\": 1658764076, \"timeNano\": 1658764076358977720}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"attach","id":"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071","from":"ubuntu","Type":"container","Action":"attach","Actor":{"ID":"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071","Attributes":{"image":"ubuntu","name":"charming_euler"}},"scope":"local","time":"1658764076","timeNano":"1658764076358977792.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T15:47:56.397+0000","rule":{"level":3,"description":"Docker: Network bridge connected","id":"87928","firedtimes":1,"mail":false,"groups":["docker"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764076.1247355","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"Type\": \"network\", \"Action\": \"connect\", \"Actor\": {\"ID\": \"e32c4d4eea1d7ccf5379bf4a5efcd5531b0dddff0178a7c6df2233c96d64e8af\", \"Attributes\": {\"container\": \"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071\", \"name\": \"bridge\", \"type\": \"bridge\"}}, \"scope\": \"local\", \"time\": 1658764076, \"timeNano\": 1658764076393986234}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"Type":"network","Action":"connect","Actor":{"ID":"e32c4d4eea1d7ccf5379bf4a5efcd5531b0dddff0178a7c6df2233c96d64e8af","Attributes":{"container":"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071","name":"bridge","type":"bridge"}},"scope":"local","time":"1658764076","timeNano":"1658764076393986304.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T15:47:56.730+0000","rule":{"level":3,"description":"Docker: Container charming_euler started","id":"87903","firedtimes":1,"mail":false,"groups":["docker"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764076.1248284","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"start\", \"id\": \"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"start\", \"Actor\": {\"ID\": \"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"charming_euler\"}}, \"scope\": \"local\", \"time\": 1658764076, \"timeNano\": 1658764076729364199}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"start","id":"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071","from":"ubuntu","Type":"container","Action":"start","Actor":{"ID":"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071","Attributes":{"image":"ubuntu","name":"charming_euler"}},"scope":"local","time":"1658764076","timeNano":"1658764076729364224.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T15:47:56.799+0000","rule":{"level":7,"description":"Docker: Container charming_euler received the action: die","id":"87924","firedtimes":1,"mail":false,"groups":["docker"],"gdpr":["IV_32.2"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764076.1249286","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"die\", \"id\": \"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"die\", \"Actor\": {\"ID\": \"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071\", \"Attributes\": {\"exitCode\": \"0\", \"image\": \"ubuntu\", \"name\": \"charming_euler\"}}, \"scope\": \"local\", \"time\": 1658764076, \"timeNano\": 1658764076798659023}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"die","id":"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071","from":"ubuntu","Type":"container","Action":"die","Actor":{"ID":"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071","Attributes":{"exitCode":"0","image":"ubuntu","name":"charming_euler"}},"scope":"local","time":"1658764076","timeNano":"1658764076798659072.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T15:47:56.961+0000","rule":{"level":4,"description":"Docker: Network bridge disconnected","id":"87929","firedtimes":1,"mail":false,"groups":["docker"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764076.1250363","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"Type\": \"network\", \"Action\": \"disconnect\", \"Actor\": {\"ID\": \"e32c4d4eea1d7ccf5379bf4a5efcd5531b0dddff0178a7c6df2233c96d64e8af\", \"Attributes\": {\"container\": \"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071\", \"name\": \"bridge\", \"type\": \"bridge\"}}, \"scope\": \"local\", \"time\": 1658764076, \"timeNano\": 1658764076959676645}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"Type":"network","Action":"disconnect","Actor":{"ID":"e32c4d4eea1d7ccf5379bf4a5efcd5531b0dddff0178a7c6df2233c96d64e8af","Attributes":{"container":"392fe7e30c5d7ffd8a795c0cb01138e04803b87638b7f39b08336a45bc5d1071","name":"bridge","type":"bridge"}},"scope":"local","time":"1658764076","timeNano":"1658764076959676672.000000"}},"location":"Wazuh-Docker"}
</details>

**Run the same image, this time having the image in cache**

<details><summary>archives.json output</summary>

    {"timestamp":"2022-07-25T15:50:01.144+0000","rule":{"level":3,"description":"Docker: Container clever_nobel created","id":"87901","firedtimes":2,"mail":false,"groups":["docker"],"pci_dss":["10.2.7"],"hipaa":["164.312.b"],"nist_800_53":["AU.14"],"tsc":["CC6.8","CC7.2","CC7.3"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764201.1251301","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"create\", \"id\": \"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"create\", \"Actor\": {\"ID\": \"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"clever_nobel\"}}, \"scope\": \"local\", \"time\": 1658764201, \"timeNano\": 1658764201143235380}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"create","id":"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e","from":"ubuntu","Type":"container","Action":"create","Actor":{"ID":"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e","Attributes":{"image":"ubuntu","name":"clever_nobel"}},"scope":"local","time":"1658764201","timeNano":"1658764201143235328.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T15:50:01.189+0000","rule":{"level":3,"description":"Docker: Attached local standard input, output, and error streams to container clever_nobel","id":"87922","firedtimes":2,"mail":false,"groups":["docker"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764201.1253309","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"attach\", \"id\": \"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"attach\", \"Actor\": {\"ID\": \"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"clever_nobel\"}}, \"scope\": \"local\", \"time\": 1658764201, \"timeNano\": 1658764201144391386}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"attach","id":"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e","from":"ubuntu","Type":"container","Action":"attach","Actor":{"ID":"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e","Attributes":{"image":"ubuntu","name":"clever_nobel"}},"scope":"local","time":"1658764201","timeNano":"1658764201144391424.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T15:50:01.189+0000","rule":{"level":3,"description":"Docker: Network bridge connected","id":"87928","firedtimes":2,"mail":false,"groups":["docker"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764201.1253309","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"Type\": \"network\", \"Action\": \"connect\", \"Actor\": {\"ID\": \"e32c4d4eea1d7ccf5379bf4a5efcd5531b0dddff0178a7c6df2233c96d64e8af\", \"Attributes\": {\"container\": \"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e\", \"name\": \"bridge\", \"type\": \"bridge\"}}, \"scope\": \"local\", \"time\": 1658764201, \"timeNano\": 1658764201164982938}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"Type":"network","Action":"connect","Actor":{"ID":"e32c4d4eea1d7ccf5379bf4a5efcd5531b0dddff0178a7c6df2233c96d64e8af","Attributes":{"container":"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e","name":"bridge","type":"bridge"}},"scope":"local","time":"1658764201","timeNano":"1658764201164983040.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T15:50:01.382+0000","rule":{"level":3,"description":"Docker: Container clever_nobel started","id":"87903","firedtimes":2,"mail":false,"groups":["docker"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764201.1254361","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"start\", \"id\": \"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"start\", \"Actor\": {\"ID\": \"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"clever_nobel\"}}, \"scope\": \"local\", \"time\": 1658764201, \"timeNano\": 1658764201382013758}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"start","id":"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e","from":"ubuntu","Type":"container","Action":"start","Actor":{"ID":"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e","Attributes":{"image":"ubuntu","name":"clever_nobel"}},"scope":"local","time":"1658764201","timeNano":"1658764201382013696.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T15:50:01.417+0000","rule":{"level":7,"description":"Docker: Container clever_nobel received the action: die","id":"87924","firedtimes":2,"mail":false,"groups":["docker"],"gdpr":["IV_32.2"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764201.1255357","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"die\", \"id\": \"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"die\", \"Actor\": {\"ID\": \"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e\", \"Attributes\": {\"exitCode\": \"0\", \"image\": \"ubuntu\", \"name\": \"clever_nobel\"}}, \"scope\": \"local\", \"time\": 1658764201, \"timeNano\": 1658764201416602202}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"die","id":"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e","from":"ubuntu","Type":"container","Action":"die","Actor":{"ID":"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e","Attributes":{"exitCode":"0","image":"ubuntu","name":"clever_nobel"}},"scope":"local","time":"1658764201","timeNano":"1658764201416602112.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T15:50:01.590+0000","rule":{"level":4,"description":"Docker: Network bridge disconnected","id":"87929","firedtimes":2,"mail":false,"groups":["docker"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764201.1256428","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"Type\": \"network\", \"Action\": \"disconnect\", \"Actor\": {\"ID\": \"e32c4d4eea1d7ccf5379bf4a5efcd5531b0dddff0178a7c6df2233c96d64e8af\", \"Attributes\": {\"container\": \"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e\", \"name\": \"bridge\", \"type\": \"bridge\"}}, \"scope\": \"local\", \"time\": 1658764201, \"timeNano\": 1658764201587939248}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"Type":"network","Action":"disconnect","Actor":{"ID":"e32c4d4eea1d7ccf5379bf4a5efcd5531b0dddff0178a7c6df2233c96d64e8af","Attributes":{"container":"066479c7e13dfe144d5f83d7d87e154e560748a16813adb36e070492e6ddc69e","name":"bridge","type":"bridge"}},"scope":"local","time":"1658764201","timeNano":"1658764201587939328.000000"}},"location":"Wazuh-Docker"}

</details>

**Remove the image**

<details><summary>archives.json output</summary>

    {"timestamp":"2022-07-25T15:52:06.530+0000","rule":{"level":3,"description":"Docker: Image sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee untagged","id":"87919","firedtimes":1,"mail":false,"groups":["docker"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764326.1257366","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"untag\", \"id\": \"sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee\", \"Type\": \"image\", \"Action\": \"untag\", \"Actor\": {\"ID\": \"sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee\", \"Attributes\": {\"name\": \"sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee\"}}, \"scope\": \"local\", \"time\": 1658764326, \"timeNano\": 1658764326528283811}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"untag","id":"sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee","Type":"image","Action":"untag","Actor":{"ID":"sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee","Attributes":{"name":"sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee"}},"scope":"local","time":"1658764326","timeNano":"1658764326528283904.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T15:52:06.536+0000","rule":{"level":7,"description":"Docker: Container sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee deleted","id":"87921","mitre":{"id":["T1070.004","T1561.001"],"tactic":["Defense Evasion","Impact"],"technique":["File Deletion","Disk Content Wipe"]},"firedtimes":1,"mail":false,"groups":["docker"],"pci_dss":["10.2.7","11.5"],"hipaa":["164.312.b","164.312.c.1","164.312.c.2"],"nist_800_53":["AU.14","SI.7"],"tsc":["CC6.8","CC7.2","CC7.3","PI1.4","PI1.5","CC6.1"]},"agent":{"id":"001","name":"75098a5c23f2","ip":"172.23.0.5"},"manager":{"name":"fe3070410846"},"id":"1658764326.1258461","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"delete\", \"id\": \"sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee\", \"Type\": \"image\", \"Action\": \"delete\", \"Actor\": {\"ID\": \"sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee\", \"Attributes\": {\"name\": \"sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee\"}}, \"scope\": \"local\", \"time\": 1658764326, \"timeNano\": 1658764326533241019}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"delete","id":"sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee","Type":"image","Action":"delete","Actor":{"ID":"sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee","Attributes":{"name":"sha256:27941809078cc9b2802deb2b0bb6feed6c236cde01e487f200e24653533701ee"}},"scope":"local","time":"1658764326","timeNano":"1658764326533241088.000000"}},"location":"Wazuh-Docker"}
</details>

**Execute an application in a container**
Run `docker run -td ubuntu bash`

<details><summary>archives.json output</summary>

    {"timestamp":"2022-07-25T16:13:25.832+0000","rule":{"level":3,"description":"Docker: Image or repository ubuntu pulled","id":"87932","firedtimes":1,"mail":false,"groups":["docker"],"pci_dss":["10.2.7"],"hipaa":["164.312.b"],"nist_800_53":["AU.14"],"tsc":["CC6.8","CC7.2","CC7.3"]},"agent":{"id":"001","name":"b041b633de29","ip":"172.24.0.5"},"manager":{"name":"03e80b7cd1a9"},"id":"1658765605.1244166","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"pull\", \"id\": \"ubuntu:latest\", \"Type\": \"image\", \"Action\": \"pull\", \"Actor\": {\"ID\": \"ubuntu:latest\", \"Attributes\": {\"name\": \"ubuntu\"}}, \"scope\": \"local\", \"time\": 1658765605, \"timeNano\": 1658765605832295217}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"pull","id":"ubuntu:latest","Type":"image","Action":"pull","Actor":{"ID":"ubuntu:latest","Attributes":{"name":"ubuntu"}},"scope":"local","time":"1658765605","timeNano":"1658765605832295168.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T16:13:25.842+0000","rule":{"level":3,"description":"Docker: Container crazy_lederberg created","id":"87901","firedtimes":1,"mail":false,"groups":["docker"],"pci_dss":["10.2.7"],"hipaa":["164.312.b"],"nist_800_53":["AU.14"],"tsc":["CC6.8","CC7.2","CC7.3"]},"agent":{"id":"001","name":"b041b633de29","ip":"172.24.0.5"},"manager":{"name":"03e80b7cd1a9"},"id":"1658765605.1244921","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"create\", \"id\": \"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"create\", \"Actor\": {\"ID\": \"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"crazy_lederberg\"}}, \"scope\": \"local\", \"time\": 1658765605, \"timeNano\": 1658765605841931944}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"create","id":"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343","from":"ubuntu","Type":"container","Action":"create","Actor":{"ID":"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343","Attributes":{"image":"ubuntu","name":"crazy_lederberg"}},"scope":"local","time":"1658765605","timeNano":"1658765605841932032.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T16:13:25.905+0000","rule":{"level":3,"description":"Docker: Network bridge connected","id":"87928","firedtimes":1,"mail":false,"groups":["docker"]},"agent":{"id":"001","name":"b041b633de29","ip":"172.24.0.5"},"manager":{"name":"03e80b7cd1a9"},"id":"1658765605.1246009","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"Type\": \"network\", \"Action\": \"connect\", \"Actor\": {\"ID\": \"e32c4d4eea1d7ccf5379bf4a5efcd5531b0dddff0178a7c6df2233c96d64e8af\", \"Attributes\": {\"container\": \"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343\", \"name\": \"bridge\", \"type\": \"bridge\"}}, \"scope\": \"local\", \"time\": 1658765605, \"timeNano\": 1658765605856166934}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"Type":"network","Action":"connect","Actor":{"ID":"e32c4d4eea1d7ccf5379bf4a5efcd5531b0dddff0178a7c6df2233c96d64e8af","Attributes":{"container":"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343","name":"bridge","type":"bridge"}},"scope":"local","time":"1658765605","timeNano":"1658765605856166912.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T16:13:26.108+0000","rule":{"level":3,"description":"Docker: Container crazy_lederberg started","id":"87903","firedtimes":1,"mail":false,"groups":["docker"]},"agent":{"id":"001","name":"b041b633de29","ip":"172.24.0.5"},"manager":{"name":"03e80b7cd1a9"},"id":"1658765606.1246938","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"start\", \"id\": \"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"start\", \"Actor\": {\"ID\": \"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"crazy_lederberg\"}}, \"scope\": \"local\", \"time\": 1658765606, \"timeNano\": 1658765606107834289}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"start","id":"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343","from":"ubuntu","Type":"container","Action":"start","Actor":{"ID":"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343","Attributes":{"image":"ubuntu","name":"crazy_lederberg"}},"scope":"local","time":"1658765606","timeNano":"1658765606107834368.000000"}},"location":"Wazuh-Docker"}

</details>

**Kill the container previously launched**
Run `docker kill $(docker ps -q)`.

<details><summary>archives.json output</summary>

    {"timestamp":"2022-07-25T16:15:18.952+0000","rule":{"level":7,"description":"Docker: Container crazy_lederberg received the action: kill","id":"87924","firedtimes":1,"mail":false,"groups":["docker"],"gdpr":["IV_32.2"]},"agent":{"id":"001","name":"b041b633de29","ip":"172.24.0.5"},"manager":{"name":"03e80b7cd1a9"},"id":"1658765718.1247943","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"kill\", \"id\": \"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"kill\", \"Actor\": {\"ID\": \"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"crazy_lederberg\", \"signal\": \"9\"}}, \"scope\": \"local\", \"time\": 1658765718, \"timeNano\": 1658765718950834779}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"kill","id":"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343","from":"ubuntu","Type":"container","Action":"kill","Actor":{"ID":"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343","Attributes":{"image":"ubuntu","name":"crazy_lederberg","signal":"9"}},"scope":"local","time":"1658765718","timeNano":"1658765718950834688.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T16:15:19.004+0000","rule":{"level":7,"description":"Docker: Container crazy_lederberg received the action: die","id":"87924","firedtimes":2,"mail":false,"groups":["docker"],"gdpr":["IV_32.2"]},"agent":{"id":"001","name":"b041b633de29","ip":"172.24.0.5"},"manager":{"name":"03e80b7cd1a9"},"id":"1658765719.1249024","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"die\", \"id\": \"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"die\", \"Actor\": {\"ID\": \"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343\", \"Attributes\": {\"exitCode\": \"137\", \"image\": \"ubuntu\", \"name\": \"crazy_lederberg\"}}, \"scope\": \"local\", \"time\": 1658765719, \"timeNano\": 1658765719003197828}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"die","id":"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343","from":"ubuntu","Type":"container","Action":"die","Actor":{"ID":"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343","Attributes":{"exitCode":"137","image":"ubuntu","name":"crazy_lederberg"}},"scope":"local","time":"1658765719","timeNano":"1658765719003197952.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2022-07-25T16:15:19.165+0000","rule":{"level":4,"description":"Docker: Network bridge disconnected","id":"87929","firedtimes":1,"mail":false,"groups":["docker"]},"agent":{"id":"001","name":"b041b633de29","ip":"172.24.0.5"},"manager":{"name":"03e80b7cd1a9"},"id":"1658765719.1250108","cluster":{"name":"wazuh","node":"wd_master_1"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"Type\": \"network\", \"Action\": \"disconnect\", \"Actor\": {\"ID\": \"e32c4d4eea1d7ccf5379bf4a5efcd5531b0dddff0178a7c6df2233c96d64e8af\", \"Attributes\": {\"container\": \"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343\", \"name\": \"bridge\", \"type\": \"bridge\"}}, \"scope\": \"local\", \"time\": 1658765719, \"timeNano\": 1658765719163671680}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"Type":"network","Action":"disconnect","Actor":{"ID":"e32c4d4eea1d7ccf5379bf4a5efcd5531b0dddff0178a7c6df2233c96d64e8af","Attributes":{"container":"e85fd22f8938f0e3ceb486f1baf0e72d70a844e6e628461713e59f63987e7343","name":"bridge","type":"bridge"}},"scope":"local","time":"1658765719","timeNano":"1658765719163671552.000000"}},"location":"Wazuh-Docker"}

</details>
